File: 150. Evaluate Reverse Polish Notation/solution_classic.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        ops = ['+', '-', '*', '/']
        
        for token in tokens:
            if token not in ops:
                stack.append(int(token))
            else:
                rightop, leftop = stack.pop(), stack.pop()
                if token == '+':
                    stack.append(leftop + rightop)
                elif token == '-':
                    stack.append(leftop - rightop)
                elif token == '*':
                    stack.append(leftop * rightop)
                elif token == '/':
                    stack.append(int(leftop / rightop))
        
        return stack[0]
                

File: 150. Evaluate Reverse Polish Notation/solution_classic.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<int> stack;
        string ops = "+-*/";
        for (auto const& token: tokens) {
            cout << token;
            if (ops.find(token) == string::npos) {
                stack.push_back(stoi(token));
            } else {
                int right_op = stack.back();
                stack.pop_back();
                int left_op = stack.back();
                stack.pop_back();
                
                if (token == "+") {
                    stack.push_back(left_op + right_op); 
                } else if (token == "-") {
                    stack.push_back(left_op - right_op); 
                } else if (token == "*") {
                    stack.push_back(left_op * right_op); 
                } else if (token == "/") {
                    stack.push_back(left_op / right_op); 
                } 
                    
            }
        }
        
        return stack[0];
    }
};
